rtt/os/gnulinux/fosi_internal.cpp

```diff
@@ -254,9 +254,16 @@ namespace RTT
 
         if (task->wait_policy == ORO_WAIT_ABS)
         {
+          // in the case of overrun by more than 4 periods,
+          // skip all the updates before now, with the next update aligned to period
+          int maxDelayInPeriods = 4;
+          NANO_TIME period = task->period;
+          if (now - wake > maxDelayInPeriods*period) {
+            period = period * ((now - wake) / period);
```

Comment:
Actually this should have been
```
period += period * ((now - wake) / period);
```
because we always want to add at least one period ([line 260](https://github.com/orocos-toolchain/rtt/pull/129/files#diff-2f91e9fcda1ad4baa9b66f9cc15b9c88R260)) plus eventually the additional ones, like in [Timer.cpp:108](https://github.com/orocos-toolchain/rtt/pull/129/files#diff-8741c88e5586ddedd496eba04289892eR108).